src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/MobEventsUtils.java

```diff
@@ -0,0 +1,221 @@
+            numberOfChiefs = equalizeBarbarianSpawns(hordeTotal, numberOfChiefs);
+        }
+    }
+
+    /**
+     * Calculate a random spawn point along the colony's border
+     *
+     * @param world  in the world.
+     * @param colony the Colony to spawn the barbarians near.
+     * @return Returns the random blockPos
+     */
+    private static BlockPos calculateSpawnLocation(final World world, final Colony colony)
+    {
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(world, colony.getCenter());
+        if (colonyView == null)
+        {
+            return null;
+        }
+        final BlockPos center = colonyView.getCenter();
+        final int radius = Configurations.workingRangeTownHall;
+
+        final int randomDegree = world.rand.nextInt((int) WHOLE_CIRCLE);
+
+        final double rads = (double) randomDegree / HALF_A_CIRCLE * Math.PI;
+        final double x = Math.round(center.getX() + radius * Math.sin(rads));
+        final double z = Math.round(center.getZ() + radius * Math.cos(rads));
+        return world.getTopSolidOrLiquidBlock(new BlockPos(x, center.getY(), z));
+    }
+
+    /**
+     * Takes a colony and spits out that colony's RaidLevel.
+     *
+     * @param colony The colony to use
+     * @return an int describing the raid level
+     */
+    public static int getColonyRaidLevel(final Colony colony)
+    {
+        int levels = 0;
+
+        @NotNull final List<CitizenData> citizensList = new ArrayList<>();
+        citizensList.addAll(colony.getCitizens().values());
+
+        for (@NotNull final CitizenData citizen : citizensList)
+        {
+            if (citizen.getJob() != null && citizen.getWorkBuilding() != null)
+            {
+                final int buildingLevel = citizen.getWorkBuilding().getBuildingLevel();
+                levels += buildingLevel;
+            }
+        }
+
+        if (colony.getTownHall() != null)
+        {
+            return (levels + colony.getTownHall().getBuildingLevel());
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    /**
+     * Reduces barbarian spawns to less than the maximum allowed (set via the config)
+     *
+     * @param total    The current horde size
+     * @param numberOf The number of barbarians which we are reducing
+     * @return the new number that the barbarians should be set to.
+     */
+    private static int equalizeBarbarianSpawns(final int total, final int numberOf)
+    {
+        int returnValue = numberOf;
+        if (total > MAX_SIZE)
+        {
+            returnValue = total - MAX_SIZE;
+
+            if (returnValue < 0)
+            {
+                return 0;
+            }
+            return returnValue;
+        }
+        return returnValue;
+    }
+
+    public static boolean isItTimeToRaid(final World world, final Colony colony)
+    {
+        if (world.getWorldTime() % 12000 == 0)
+        {
+            TimeToRaid = world.getWorldTime() + 1000;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1000 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)